src/app/switcher/switcher.service.ts

```diff
@@ -76,23 +80,78 @@ export class SwitcherService {
     private sharedService: SharedService,
     private pusherService: PusherService,
     private reviewsService: ReviewListService,
-    private eventsService: EventListService,
+    private eventsService: EventListService
   ) {}
 
-  getPrograms() {
-    const programs = this.storage.get('programs') || [];
+  getExperience(stackList: Stack[]) {
+    if (!this.storage.stacks || this.storage.stacks.length < 1) {
+      throw Error('Fail to retrieve stacks info');
+    }
+    const stackRequests = [];
+    const apikeyFromLoginAPI = this.storage.loginApiKey;
+    const body = new HttpParams()
+      .set('apikey', apikeyFromLoginAPI);
+    const headers = {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      service: 'LOGIN'
+    };
+    stackList.forEach(stack => {
+      stackRequests.push(this.request.post(
+        {
+          endPoint: this.utils.urlFormatter(stack.coreApi, api.login),
+          data: body,
+          httpOptions: { headers },
+          isFullUrl: true
+        }).pipe(map(res => {
+          res.stack = stack;
+          return res;
+        })));
+    });
+    return forkJoin(stackRequests).pipe(map(res => this._normaliseAuthResults(res)));
+  }
+
+  private _normaliseAuthResults(apiResults: any[]): any {
     const cdn = 'https://cdn.filestackcontent.com/resize=fit:crop,width:';
     let imagewidth = 600;
-    programs.forEach(program => {
-      if (program.project.lead_image) {
-        const imageId = program.project.lead_image.split('/').pop();
-        if (!this.utils.isMobile()) {
-          imagewidth = 1024;
+    const expreancesList = [];
+    apiResults.forEach(result => {
+      const data = result.data;
+      data.Timelines.map(
```

Comment:
@sasangachathumal I'm not sure if this possible, do we need to cover condition where the  `Timelines` object from data is null or undefined?

Reply:
i will check, better to have something like that. i will check and add.